### DISCUSSION TITLE
Vaillant ecoTEC plus - AdapterV5 - Homeassistant

### DISCUSSION THREAD

**Discussion by gentleman1979** [john30/ebusd#1200] [Q&A]
Hello! 
I have a Vaillant heating system and have successfully set up the Adapter-V5. I receive many values, for example:

`bai
SetMode = { "hcmode": {"value": "auto"}, "flowtempdesired": {"value": 90.0}, "hwctempdesired": {"value": 65.0}, "hwcflowtempdesired": {"value": null}, "disablehc": {"value": 0}, "disablehwctapping": {"value": 0}, "disablehwcload": {"value": 1}, "remoteControlHcPump": {"value": 0}, "releaseBackup": {"value": 0}, "releaseCooling": {"value": 0}}
StatusCirPump = { "0": {"name": "", "value": "on"}}
Status01 = { "0": {"name": "temp1", "value": 72.0}, "1": {"name": "temp1", "value": 70.5}, "2": {"name": "temp2", "value": null}, "3": {"name": "temp1", "value": 55.0}, "4": {"name": "temp1", "value": 61.5}, "5": {"name": "pumpstate", "value": "off"}}
ReturnTempExternal = { "temp": {"value": -1.81}, "sensor": {"value": "cutoff"}}
ReturnTempMax = { "temp": {"value": 87.25}}
DateTime = { "dcfstate": {"value": "nosignal"}, "btime": {"value": "-:-:-"}, "bdate": {"value": "-.-.-"}, "temp2": {"value": null}}
Status02 = { "0": {"name": "hwcmode", "value": "on"}, "1": {"name": "temp0", "value": 60}, "2": {"name": "temp1", "value": 75.0}, "3": {"name": "temp0", "value": 70}, "4": {"name": "temp1", "value": 65.0}}
SHEMaxFlowTemp = { "temp": {"value": 80.56}}
StorageExitTemp = { "temp": {"value": 116.06}, "sensor": {"value": "circuit"}}
StorageLoadPumpHours = { "hoursum2": {"value": 26}}
StorageloadPumpStarts = { "cntstarts2": {"value": 1321}}
StorageTemp = { "temp": {"value": 61.56}, "sensor": {"value": "ok"}}
StorageTempDesired = { "temp": {"value": 60.00}}
StorageTempMax = { "temp": {"value": 70.25}}
TempDiffBlock = { "temp0": {"value": 0}}
TempDiffFailure = { "temp0": {"value": 0}}
TempGradientFailure = { "temp0": {"value": 0}}
Templimiter = { "0": {"name": "", "value": "off"}}
TemplimiterWithNTC = { "0": {"name": "", "value": "yes"}}
TempMaxDiffExtTFT = { "temp": {"value": 0.00}}
ValveStarts = { "cntstarts2": {"value": 3304}}
WaterpressureMeasureCounter = { "0": {"name": "", "value": 131}}
WPPostrunTime = { "minutes0": {"value": 5}}
CounterStartattempts1 = { "temp0": {"value": 1}}
CounterStartattempts2 = { "temp0": {"value": 0}}
CounterStartAttempts3 = { "temp0": {"value": 0}}
CounterStartAttempts4 = { "temp0": {"value": 0}}
DeactivationsTemplimiter = { "0": {"name": "", "value": 0}}
DisplayMode = { "0": {"name": "", "value": 2}}
ExtFlowTempDesiredMin = { "temp": {"value": 40.00}}
FanHours = { "hoursum2": {"value": 1164}}
FanStarts = { "cntstarts2": {"value": 0}}
FlowTemp = { "temp": {"value": 79.00}, "sensor": {"value": "ok"}}
FlowTempDesired = { "temp": {"value": 75.00}}
FlowTempMax = { "temp": {"value": 87.62}}
HcHours = { "hoursum2": {"value": 1089}}
HcPumpMode = { "0": {"name": "", "value": 3}}
HcPumpStarts = { "cntstarts2": {"value": 21532}}
HcStarts = { "0": {"name": "", "value": 28400}}
HcUnderHundredStarts = { "0": {"name": "", "value": 6}}
HoursTillService = { "hoursum2": {"value": null}}
HwcHours = { "hoursum2": {"value": 20}}
HwcStarts = { "0": {"name": "", "value": 800}}
HwcTemp = { "temp": {"value": 54.38}, "sensor": {"value": "ok"}}
HwcTempMax = { "temp": {"value": 65.00}}
HwcUnderHundredStarts = { "0": {"name": "", "value": 72}}
maintenancedata_HwcTempMax = { "temp": {"value": 78.56}}
ModulationTempDesired = { "0": {"name": "", "value": 16.0}}
OutdoorstempSensor = { "temp": {"value": -60.44}, "sensor": {"value": "cutoff"}}
OverflowCounter = { "yesno": {"value": "yes"}}
PowerValue = { "0": {"name": "", "value": "10 04 53 14 64 18"}}
PrAPSCounter = { "0": {"name": "", "value": 0}}
PrEnergyCountHc1 = { "0": {"name": "", "value": 3948334}}
PrEnergyCountHc2 = { "0": {"name": "", "value": 0}}
PrEnergyCountHc3 = { "0": {"name": "", "value": 0}}
PrEnergyCountHwc1 = { "0": {"name": "", "value": 51065}}
PrEnergyCountHwc2 = { "0": {"name": "", "value": 0}}
PrEnergyCountHwc3 = { "0": {"name": "", "value": 0}}
PrEnergySumHc1 = { "0": {"name": "", "value": 514871332}}
PrEnergySumHc2 = { "0": {"name": "", "value": 0}}
PrEnergySumHc3 = { "0": {"name": "", "value": 0}}
PrEnergySumHwc1 = { "0": {"name": "", "value": 16911012}}
PrEnergySumHwc2 = { "0": {"name": "", "value": 0}}
PrEnergySumHwc3 = { "0": {"name": "", "value": 0}}
PrVortexFlowSensorValue = { "0": {"name": "", "value": 0}}
PumpHours = { "hoursum2": {"value": 2497}}
PumpPower = { "0": {"name": "", "value": 15}}
PumpPowerDesired = { "0": {"name": "", "value": "auto"}}
ReturnTemp = { "temp": {"value": 79.81}, "tempmirror": {"value": 64258}, "sensor": {"value": "ok"}}
broadcast
vdatetime = { "time": {"value": "21:25:57"}, "date": {"value": "20.02.2024"}}
350
ActualTempDesired = { "temp0": {"value": 0}}
ComfTempEnabled = { "onoff": {"value": "on"}}
ControlMode = { "0": {"name": "", "value": -82}}
FillmodeStartTime = { "0": {"name": "", "value": 1}}
FlowTempMin = { "temp": {"value": 0.00}}
HwcTempDesired = { "temp": {"value": 0.00}}
IsInSingleHwcLoadingMode = { "onoff": {"value": "off"}}
OffsetDesTemp = { "temp": {"value": 16.06}}
OperatingMode = { "mcmode": {"value": 227}}
OperatingModeHwc = { "mcmode": {"value": "on"}}
OperatingmodeStartTime = { "0": {"name": "", "value": 0}}
OutsideTemp = { "temp": {"value": 33.44}, "sensor": {"value": 20}}
PrevOperatingMode = { "mcmode": {"value": 32}}
PumpPower = { "power": {"value": 1}}
ResetYield = { "onoff": {"value": 7}}
RestoreOpModeAfterHoliday = { "onoff": {"value": "on"}}
resvdColl1Sensor = { "temp": {"value": 1590.19}, "sensor": {"value": 99}}
resvdStorage1Sensor = { "temp": {"value": 21.38}, "sensor": {"value": "ok"}}
RoomTempOffset = { "temp": {"value": 33.44}}
RoomTempSwitchOn = { "0": {"name": "", "value": "modulating"}}
StatusDcf = { "dcfstate": {"value": "ok"}}
TeleSwOperatingMode = { "mcmode": {"value": "eco"}}`

![homeassistant-ebusd](https://github.com/john30/ebusd/assets/121397274/37435d23-2754-472a-92f1-c6d24ce0e71d)

But can someone tell me how I switch the operating mode from 'auto' to 'frost protection' via Home Assistant?"

**Comment by Koky05** [comment]
At first you need to allow write values to be shown by edditing `filter-direction=r|u|^w`. It depend on way you run ebusd (compiled directly, docker container, HA Addon).

**Reply by gentleman1979** [reply]
Do you mean the read-only option? In Home Assistant, it is off.
![Screenshot 2024-02-20 225517](https://github.com/john30/ebusd/assets/121397274/010db9fa-68f1-46e6-b97e-1ea2fd5316d2)

**Comment by Koky05** [comment]
No in file `mqtt-hassio.cfg` which operate MQTT autodicovery options has default settings:
`filter-direction=r|u` and it is need to be changed to value `filter-direction=r|u|^w`.

Into field `Custom Command Line option` add `--mqttvar=filter-direction=r|u|^w` (variable injection method).

**Reply by gentleman1979** [reply]
I'm sorry, I still don't understand. Do I first need to fetch with 
`wget https://raw.githubusercontent.com/john30/ebusd/master/contrib/etc/ebusd/mqtt-hassio.cfg`  in the HA terminal? And then edit the file? Is it enough if I enter the lines `filter-direction=r|u|^w` und `Custom Command Line option ----mqttvar=filter-direction=r|u|^w`?

**Reply by nobbilie** [reply]
Hi,
lege dir einen Ordner in deinem Config-Pfad von Home Assistant an. Dort kopier dir die mqtt-hassio.cfg hinein.
Danach machst du das gleiche mit den Konfigurationsdateien in einem weiteren Ordner.

Das sieht dann so bei mir aus in den Konfigurationsoptionen des Addons:

![image](https://github.com/john30/ebusd/assets/154513250/766aed5b-fcd3-4e9e-8fab-fe7770b4a734)

![image](https://github.com/john30/ebusd/assets/154513250/3630d55f-28bd-4ca6-b9df-b6183901e3c9)


Und den Befehl den Koky05 meint, kopierst du dort hinein:

![image](https://github.com/john30/ebusd/assets/154513250/6075f8f3-c56b-4e33-995f-937cd860b6c2)

Die Latenz die ich noch drin habe, braucht es für die Lüftungsanlage hinter einem VR32.
Ich bin zar auch noch blutiger Anfänger was den ebusd angeht, habe aber hier schon viel dazu gelernt.

**Reply by gentleman1979** [reply]
Vielen Dank für deine Hilfe! Jetzt habe ich das auch so übernommen. Kann ich 
![image](https://github.com/john30/ebusd/assets/121397274/92697cb2-4f7e-4f40-8098-b738c677bd4f)
so lassen? Kannst du mir noch zeigen, wie ich die Heizung von "Frostschutz" auf "Automatik" umschalte?
lg

**Reply by nobbilie** [reply]
Leider nein, bin selbst da noch nicht so drin - bei mir funktioniert das alles auchnoch nicht so, wie ich es gern hätte.
